### PR TITLE
document changes to debt settlement repayment flow

### PR DESCRIPTION
**Pull request by Tom-Goji**
Document updates to settlement flow.

- add notices to `/platformApi/settlement/repayment/reference` and `/platformApi/settlement/repayment` to signal they are considered legacy.
- add new endpoint `/platformApi/settlement/investor-repayment` with required `sourceWalletId` in the body for the new flow

New endpoint docs:
image

Existing endpoints:
image
image

### CODE REVIEW

**Comment by victokoh** [comment]
This is slightly awkward. Technically the description on the very last screen is still applicable/ isn't "deprecated" per-se. That's describing the deferred repayments flow which is still 100% valid. We're just introducing another case of the real-time repayments flow, where, if you are a P2P platform, making use of this functionality - payment will no longer be expected upfront - instead the platform will be asked to provide the wallet `id` of where to source the funds from. We will then handle the process of distributing the funds from there.

**Comment by victokoh** [comment]
So if they are a non-P2P platform and a super old client still on the real-time repayments flow, that entire flow of requesting a reference first before paying up - still valid. :/ I don't think we required old clients to necessarily switch over to deferred flow..

**Comment by Tom-Goji** [comment]
At the moment we have `/platformApi/settlement/repayment` which serves 2 different flows.

- deferred repayment (where they register a reference with us, transfer the money and then we distribute triggered by their API call.)
(Capital Rise)
- real time repayment (where they register the repayments then transfer the money after they get a success response from us)
(Crowd Property and basically all P2P platforms)

As part of this work we are changing how the "real time repayment" flow works. This needs to stay in place until the deprecation window has ended at which point we should be fine to remove the "old way".

I'm seriously considering suggesting we add a new endpoint to serve the "new way" real time repayment and split these 2 divergent flows once the deprecation window is up. I can see a few pros and cons:

Pros:
- new endpoint makes it easy to see if platforms have moved over yet
- potential for simplification in the future with less "if" statements deciding which flow we're using (for now can be wired identically at the RestfulAdapter)
- cleaner API removal validation
- options for better quality metrics for the endpoint (flows have drastically different performance characteristics)
- options for load shedding by endpoint if we needed to

Cons:
- 2 endpoints that trigger repayments
- it's going to be tricky to explain which endpoint new platforms should use. 
- more endpoints to maintain

My thinking is that an endpoint which is all things to all people, turns out to be a magnet for complexity. By splitting them now,  whilst we are requiring platforms to have completed the work, we can manage some of the complexity by limiting the amount of code paths you need to follow. We might not get another shot at this in the future. 😅 

Thoughts? @badgerwithagun @victokoh @stustill

File: source/includes/debt-settlement.md.erb [hunk 1]
```diff
@@ -516,25 +521,94 @@ Content-Type: application/json
...
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+""
+```
+### Description
+Adds a repayment to an investment.
```
Comment (stustill) [inline source/includes/debt-settlement.md.erb:588]: Singular? Feels like this should be plural.

File: source/includes/debt-settlement.md.erb [hunk 2]
```diff
@@ -516,25 +521,94 @@ Content-Type: application/json
...
+    "type" : "CAPITAL"
+  } ]
+}
+```
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+""
```
Comment (stustill) [inline source/includes/debt-settlement.md.erb:585]: I thought we were returning the repayment ID now?

**Review by stustill — CHANGES_REQUESTED** [review]

File: source/includes/debt-settlement.md.erb (hunk 1 above)
Comment (Tom-Goji) [reply source/includes/debt-settlement.md.erb]: Done

File: source/includes/debt-settlement.md.erb (hunk 2 above)
Comment (Tom-Goji) [reply source/includes/debt-settlement.md.erb]: Yeah my bad. Done 👍

**Review by stustill — APPROVED** [review]

**Pull request merged**